openassessment/xblock/static/js/src/lms/oa_staff_area.js

```diff
@@ -157,108 +198,148 @@
             }
         },
 
         /**
-         Set the comment text.
-         Retrieve the comment text.
-
-         Args:
-         text (string): reason to .
-
-         Returns:
-         string: The current comment text.
-         **/
+         * Set the comment text.
+         *
+         * Retrieve the comment text.
+         *
+         * Args:
+         *   text (string): reason to .
+         *
+         * Returns:
+         *   string: The current comment text.
+         */
         comment: function(text) {
-            var sel = $('#staff-info__cancel-submission__comments', this.element);
+            var $submissionComments = $('.cancel_submission_comments', this.element);
             if (typeof text === 'undefined') {
-                return sel.val();
+                return $submissionComments.val();
             } else {
-                sel.val(text);
+                $submissionComments.val(text);
             }
         },
 
         /**
-         Enable/disable the cancel submission based on whether
-         the user has entered a comment.
-         **/
+         * Enable/disable the cancel submission based on whether
+         * the user has entered a comment.
+         */
         handleCommentChanged: function() {
             // Enable the cancel submission button only for non-blank comments
-            var isBlank = ($.trim(this.comment()) !== '');
+            var isBlank = $.trim(this.comment()) !== '';
             this.cancelSubmissionEnabled(isBlank);
+        },
+
+
+        /**
+         * Enable/disable the staff assessment submit button.
+         *
+         * @param enabled If specified, sets the state of the button.
+         * @returns {boolean} Whether the button is enabled
+         */
+        staffSubmitEnabled: function(enabled) {
+            var button = $('.wrapper--staff-assessment .action--submit', this.element);
+            if (typeof enabled === 'undefined') {
+                return !button.hasClass('is--disabled');
+            } else {
+                button.toggleClass('is--disabled', !enabled);
+            }
+        },
+
+        /**
+         * Submit the staff assessment.
+         *
+         * @param rubric The rubric being assessed.
+         */
+        submitStaffAssessment: function(submissionID, rubric) {
+            // Send the assessment to the server
+            var view = this;
+            var baseView = this.baseView;
+            baseView.toggleActionError('staff', null);
+            view.staffSubmitEnabled(false);
+
+            this.server.staffAssess(
+                rubric.optionsSelected(), rubric.criterionFeedback(), rubric.overallFeedback(), submissionID
+                )
+                .done(function() {
+                    view.loadStudentInfo({expanded_view:  'final-grade'});
+                })
+                .fail(function(errorMessage) {
+                    baseView.toggleActionError('staff', errorMessage);
```

Comment:
No. :-) I've added one, and also added a test for the failure case for canceling a submission.